src/titiler/core/titiler/core/factory.py

```diff
@@ -166,6 +177,32 @@ def url_for(self, request: Request, name: str, **path_params: Any) -> str:
             base_url += self.router_prefix.lstrip("/")
         return url_path.make_absolute_url(base_url=base_url)
 
+    def add_route_dependencies(
+        self,
+        scopes: List[EndpointScope],
+        dependencies=List[params.Depends],
+    ):
+        """Add dependencies to routes.
+
+        Allows a developer to add dependencies to a route after the route has been defined.
+
+        """
+        for route in self.router.routes:
```

Comment:
```suggestion
        for route in self.app.router.routes:
```

This is the 🐛.  Once the api is established, I think we no longer care about `self.router`.

Reply:
there is no `app` in the Factory